src/js/ValidationProgress.jsx

```diff
@@ -0,0 +1,50 @@
+var React = require('react');
+var api = require('./api');
+
+class ValidationProgress extends React.Component {
+
+  constructor(props){
```

Comment:
Mostly for fun; sharpen the saw and all that. If you'd prefer to keep it strictly ES5 we can strip it out; I just figure that since we already are leaning on babel, we might as well get some mileage from it ;)